components/color-picker/index.jsx

```diff
@@ -0,0 +1,582 @@
+	 */
+	valueWorking: PropTypes.string,
+};
+
+const defaultProps = {
+	assistiveText: {
+		saturationValueGrid:
+			'Use arrow keys to select a saturation and brightness, on an x and y axis.',
+		hueSlider: 'Select Hue',
+	},
+	events: {},
+	labels: {
+		blueAbbreviated: 'B',
+		cancelButton: 'Cancel',
+		customTab: 'Custom',
+		customTabActiveWorkingColorSwatch: 'Working Color',
+		customTabTransparentSwatch: 'Transparent',
+		greenAbbreviated: 'G',
+		hexLabel: 'Hex',
+		invalidColor: 'The color entered is invalid',
+		invalidComponent: 'The value needs to be an integer from 0-255',
+		redAbbreviated: 'R',
+		submitButton: 'Done',
+		swatchTab: 'Default',
+	},
+	menuPosition: 'absolute',
+	swatchColors: [
+		'#e3abec',
+		'#c2dbf7',
+		'#9fd6ff',
+		'#9de7da',
+		'#9df0c0',
+		'#fff099',
+		'#fed49a',
+		'#d073e0',
+		'#86baf3',
+		'#5ebbff',
+		'#44d8be',
+		'#3be282',
+		'#ffe654',
+		'#ffb758',
+		'#bd35bd',
+		'#5779c1',
+		'#5679c0',
+		'#00aea9',
+		'#3cba4c',
+		'#f5bc25',
+		'#f99221',
+		'#580d8c',
+		'#001970',
+		'#0a2399',
+		'#0b7477',
+		'#0b6b50',
+		'#b67e11',
+		'#b85d0d',
+	],
+	selectedTab: 'swatches',
+	variant: 'base',
+};
+
+class ColorPicker extends React.Component {
+	static displayName = COLOR_PICKER;
+	static propTypes = propTypes;
+	static defaultProps = defaultProps;
+
+	constructor(props) {
+		super(props);
+
+		this.generatedId = this.props.id || shortid.generate();
+		const workingColor = ColorUtils.getNewColor(
+			{
+				hex:
+					this.props.valueWorking ||
+					this.props.value ||
+					this.props.swatchColors[0],
+			},
+			this.props.events.onValidateWorkingColor
+		);
+		this.state = {
+			currentColor: this.props.value || this.props.swatchColors[0],
+			disabled: this.props.disabled,
+			isOpen: this.props.isOpen,
+			workingColor,
+			previousWorkingColor: workingColor,
+			colorErrorMessage: this.props.errorText,
+		};
+	}
+
+	componentWillMount() {
+		checkProps(COLOR_PICKER, this.props);
+	}
+
+	// use getDerivedStateFromProps when available
+	componentWillReceiveProps(nextProps) {
+		const nextState = {};
+
+		if (nextProps.value) {
+			nextState.currentColor = nextProps.value;
+		}
+
+		if (nextProps.valueWorking) {
+			nextState.workingColor = ColorUtils.getNewColor(
+				{
+					hex: nextProps.valueWorking,
+				},
+				this.props.events.onValidateWorkingColor
+			);
+		}
+
+		if (nextProps.disabled !== undefined) {
+			nextState.disabled = nextProps.disabled;
+		}
+
+		this.setState(nextState);
+	}
+
+	getInput() {
+		return this.props.hideInput ? null : (
+			<Input
+				aria-describedby={`color-picker-summary-error-${this.generatedId}`}
+				className={classNames(
+					'slds-color-picker__summary-input',
+					'slds-align-top',
+					{
+						'slds-has-error': !!this.state.colorErrorMessage,
+					}
+				)}
+				disabled={this.props.disabled}
+				id={`color-picker-summary-input-${this.generatedId}`}
+				maxLength="7"
```

Comment:
Just something to be aware of is, if we would want to support 8 digits hex, where the two last digits are for opacity, we would have to change this `maxLength`.

Reply:
SLDS doesn't have a max length. I'm OK with removing it.